test/unit/redirect.test.js

```diff
@@ -0,0 +1,138 @@
+// We override Smartcar's browser lint rules for Jest tests.
+// Note that Jest ships with jsdom so window is loaded globally in Jest tests.
+
+/* eslint strict: ['error', 'global'] */
+/* global require, expect, jest */
+
+'use strict';
+
+describe('redirect', () => {
+  beforeEach(() => {
+    // reset window.opener before each test
+    window.opener = undefined;
+    jest.resetModules();
+  });
+
+  test('throws error if window.opener undefined', () => {
+    // eslint-disable-next-line global-require
+    expect(() => require('../../src/redirect'))
+      .toThrow('window.opener must be defined');
+  });
+
+  test('calls close', () => {
+    // eslint-disable-next-line no-empty-function
+    window.opener = {postMessage: () => {}};
+
+    const mockClose = jest.fn();
+    window.close = mockClose;
+
+    require('../../src/redirect'); // eslint-disable-line global-require
+    expect(mockClose).toBeCalled();
+  });
+
+  test('calls window.postMessage & window.close', () => {
+    window.opener = {};
+
+    const mockClose = jest.fn();
+    window.close = mockClose;
+
+    const mockPostMessage = jest.fn();
+    window.opener.postMessage = mockPostMessage;
+
+    require('../../src/redirect'); // eslint-disable-line global-require
+
+    // verify postMessage and close are called on load
+    expect(mockPostMessage).toBeCalled();
+    expect(mockClose).toBeCalled();
+  });
+
+  test('should post to self hosted origin', () => {
+    window.opener = {};
+
+    const selfHostedOrigin = 'https://www.the-next-awesome-car-app.com';
+    jsdom.reconfigure({url: selfHostedOrigin}); // eslint-disable-line no-undef
+
+    const mockPost = jest.fn();
+    window.opener.postMessage = mockPost;
+
+    require('../../src/redirect'); // eslint-disable-line global-require
+    expect(mockPost).toBeCalledWith(expect.anything(), selfHostedOrigin);
```

Comment:
why do we use `expect.anything()` for the message? Thoughts on using a specific message (is that possible?)

Reply:
My thought was to test the `message` separate from the `targetOrigin`. This test and the next both test different `targetOrigin`s while the three after that test the `message`. Happy to combine these tests if people think that separating in that way is confusing or unhelpful.